float64 X
float64 Y
float64 Z
float64 MX
float64 MY
float64 MZ